public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -112,6 +115,22 @@ export const flattenRules = (rules: RuleNamespace[]) => {
   }, []);
 };
 
+export const getAlertingRule = (rule: CombinedRuleWithLocation) =>
+  isAlertingRule(rule.rule.promRule) ? rule.rule.promRule : null;
+
+export const flattenCombinedRules = (rules: CombinedRuleNamespace[]) => {
+  return rules.reduce<CombinedRuleWithLocation[]>((acc, { rulesSource, name: namespaceName, groups }) => {
+    groups.forEach(({ name: groupName, rules }) => {
+      rules.forEach((rule) => {
+        if (isAlertingRule(rule.promRule!)) {
```

Comment:
Wouldn't be better to use `rule.promRule && isAlertingRule(rule.promRule)`? This way we don't need to make unsafe assumptions about `promRule` being defined

Reply:
you're right!